text/0000-eliminate-finalize.md

```diff
@@ -0,0 +1,142 @@
+
+The address format will be `version | ed25519_key | public_excess | public_nonce` bech32 encoded using `grin` as the HRC and `1` as the version. They are easily distinguishable from the existing slatepack addresses due to their additional length (1 additional byte for version, 33 additional bytes for `public_excess` and 33 additional bytes for `public_nonce`).
+
+Example: 
+
+version: 1
+ed25519 pubkey: 0x8fe91a51badcebca96d5c8f35daf4c0419ac5312b58fa6496c361420958f90ba
+public excess: 0x03639592cc3d406a1773ead98b59aa0c54c10e6412edfcdf78c45faf6c810d90d4
+public nonce: 0x0332e9abd666ab919b22613e7f6332f449c9adb46b323b0d069953faabf15cfd45
+bech32 address: grin1qx87jxj3htwwhj5k6hy0xhd0fszpntznz26clfjfdsmpggy437gt5qmrjkfvc02qdgth86ke3dv65rz5cy8xgyhdln0h33zl4akgzrvs6spn96dt6en2hyvmyfsnulmrxt6ynjddk34nywcdq6v4874t79w063gugaa22
+
+## Payment Proofs
+
+### Building the proof
+
+Define `diffie_secret` as the Diffie-Hellman secret, which can be computed by only the sender and the receiver:
+`diffie_secret` = `Hash(sender.nonce * receiver.public_nonce)` = `Hash(sender.public_nonce * receiver.nonce)`
+
+Calculate `proof_nonce` = `Hash(diffie_secret | receiver.public_excess | tx.amount)`.
+
+The `total_nonce` for the transaction will then be `sender.nonce + receiver.nonce + proof_nonce`.
+
+* `e = SHA256(M | sender.public_nonce + receiver.public_nonce + (proof_nonce * G) | sender.public_excess + receiver.public_excess)`
+* The sender's partial signature will be `sender.partial_sig = sender.nonce + e * sender.excess`
+* The receiver's partial signature will be `receiver.partial_sig = receiver.nonce + e * receiver.excess` 
+
+The final aggregated kernel signature `(s, k*G)` can then be calculated as `(sender.partial_sig + receiver.partial_sig + proof_nonce, total_nonce * G)`
+
+### Verifying the proof
+
+The following 4 steps are necessary for the sender to prove payment to the receiver's address:
+
+1. Provide the kernel and show that it was confirmed on-chain.
+2. Prove knowledge of `sender.nonce`
+3. Show that `sender.public_nonce + receiver.public_nonce + (proof_nonce * G)` = `total_nonce * G` (ie. the `k*G` in the kernel signature)
+4. Provide the preimage to `proof_nonce` (`diffie_secret | receiver.public_excess | tx.amount`)
+
+## Slate Format
+
+A slate is now only needed for passing information from the sender to receiver. It must include the following data:
+
+* `tx.amount`
+* `tx.offset`
+* `tx.kernel_features`
+* `tx.fee`
+* `tx.inputs`
+* `receiver.change_output`
+* `receiver.public_excess` & `receiver.public_nonce` (to prevent the need for grinding)
+* `sender.public_excess`, `sender.public_nonce`, & `sender.partial_sig`
+* `time_to_live` (optional)
+
+TODO: Define JSON & binary formats
+TODO: Mention QR codes
+
+## Backward Compatibility
```

Comment:
Might also be good to outline whether we think this should be the default moving forward, and if not, i.e. if the existing flow still should be supported, how we expect the two to co-exist without added friction.

Reply:
Yea, I'm not even sure what's best to do here. I'll put some thought into it and then update this section accordingly.